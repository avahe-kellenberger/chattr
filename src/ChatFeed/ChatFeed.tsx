import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { Colors, FAB } from 'react-native-paper';
import { NavigationContainerProps } from 'react-navigation';

interface Props extends NavigationContainerProps {
  theme?: any;
}

const styles = StyleSheet.create({
  fab: {
    position: 'absolute',
    margin: 16,
    right: 0,
    bottom: 0,
  },
});
export default class ChatFeed extends React.Component<Props> {
  public constructor(props: Props) {
    super(props);
    this.navigate = this.navigate.bind(this);
  }

  public navigate() {
    if (this.props.navigation) {
      this.props.navigation.navigate('NewChat');
    }
  }

  public render() {
    return (
      <View>
        <Text>Chat Feed</Text>
        <FAB
          style={styles.fab}
          icon="add"
          theme={{ colors: { accent: Colors.red300 } }}
          onPress={this.navigate}
        />
      </View>
    );
  }
}
