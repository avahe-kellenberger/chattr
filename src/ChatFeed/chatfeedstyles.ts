import { StyleSheet } from 'react-native';

export const ChatFeedStyle = StyleSheet.create({
  view: {
    flex: 1,
    alignItems: 'center',
  },
  fab: {
    position: 'absolute',
    flex: 1,
    margin: 16,
    alignSelf: 'flex-end',
    bottom: 0,
    right: 16,
  },
});
