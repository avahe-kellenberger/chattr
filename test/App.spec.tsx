import { shallow } from 'enzyme';
import React from 'react';
import { FAB } from 'react-native-paper';
import renderer, { ReactTestRenderer } from 'react-test-renderer';
import ChatFeed from '../src/ChatFeed/ChatFeed';

test('Renders correctly with default', () => {
  const app: ReactTestRenderer = renderer.create(<ChatFeed />);
  expect(app).toMatchSnapshot();
});

const createTestProps = (props: Record<string, any>) => ({
  navigation: {
    navigate: jest.fn(),
  },
  ...props,
});

test('has a button that links to new chat', () => {
  let props: any;
  props = createTestProps({});

  const app = shallow(<ChatFeed {...props} />);

  app.find(FAB).simulate('press');
  expect(props.navigation.navigate).toBeCalledWith('NewChat');
});
