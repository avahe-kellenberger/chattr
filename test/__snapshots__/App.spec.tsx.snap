// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly with default 1`] = `
<View>
  <Text>
    Chat Feed
  </Text>
  <View
    pointerEvents="auto"
    style={
      Object {
        "backgroundColor": "#e57373",
        "borderRadius": 28,
        "bottom": 0,
        "elevation": 6,
        "margin": 16,
        "opacity": 1,
        "position": "absolute",
        "right": 0,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 5,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 6,
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 28,
          },
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            Object {
              "height": 56,
              "width": 56,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "height": 24,
                "width": 24,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "rotate": "0deg",
                  },
                ],
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
